### PR TITLE
fix parse arrays with const length correctly

### LINKED ISSUE #294: Panic on generating mock for interface with method returing array with const as size
I have the following file:
```go
package main

const C = 2

type I interface{
    F() [C]int
}
```

when I'm trying to generate mock for the `I` interface, I'm getting a panic from `mockgen`:
```go
$ mockgen -source=test.go
panic: interface conversion: ast.Expr is *ast.Ident, not *ast.BasicLit

goroutine 1 [running]:
main.(*fileParser).parseType(0xc00014fcc8, 0xc000145a90, 0xb, 0x782660, 0xc0000e8630, 0x0, 0x0, 0x3e, 0x8)
        /home/jakub/go/src/github.com/golang/mock/mockgen/parse.go:357 +0x12a9
main.(*fileParser).parseFieldList(0xc00014fcc8, 0xc000145a90, 0xb, 0xc0001a6090, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
        /home/jakub/go/src/github.com/golang/mock/mockgen/parse.go:333 +0x224
main.(*fileParser).parseFunc(0xc00014fcc8, 0xc000145a90, 0xb, 0xc0000ee340, 0xc0001b0180, 0xc00014f8a0, 0x40b5d9, 0xc0000ec380, 0xc00014f8b8, 0x40b5d9, ...)
        /home/jakub/go/src/github.com/golang/mock/mockgen/parse.go:310 +0x2c9
main.(*fileParser).parseInterface(0xc00014fcc8, 0x953ee9, 0x1, 0xc000145a90, 0xb, 0xc0000ee360, 0xc00014fa98, 0x405fd8, 0x6d9d80)
        /home/jakub/go/src/github.com/golang/mock/mockgen/parse.go:234 +0x213
main.(*fileParser).parseFile(0xc00014fcc8, 0xc000145a90, 0xb, 0xc0000f4380, 0x0, 0x0, 0xc0000f4380)
        /home/jakub/go/src/github.com/golang/mock/mockgen/parse.go:182 +0x436
main.parseFile(0x7fff90dd11e0, 0x7, 0x1, 0x1, 0x0)
        /home/jakub/go/src/github.com/golang/mock/mockgen/parse.go:100 +0x7a7
main.main()
        /home/jakub/go/src/github.com/golang/mock/mockgen/mockgen.go:65 +0xa67
```
I'm using the latest version of `mockgen` - commit: `d74b93584564161b2de771089ee697f07d8bd5b5`

### PR DESCRIPTION
**Pull request by fsmiamoto**
Fixes #294 

**Description**
As described in the related issue - #294 - parsing an array with a const defined size would generate a panic.

This PR attempts fix that issue.

Would love to hear some feedback :slightly_smiling_face: 

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes tests

**Reviewer Notes**

- [ ] The code flow looks good.
- [ ] Tests added.

### CODE REVIEW

File: mockgen/internal/tests/const_array_length/input.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+//go:generate mockgen -package const_length -destination mock.go -source input.go
```
Comment (codyoss) [inline mockgen/internal/tests/const_array_length/input.go:1]: nit: could you move this line below the package declaration. Having it here will have it show up in godoc. 

I know it is done this way in other packages, but I would like to refactor these sometime.

File: mockgen/internal/tests/const_array_length/input.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+package const_length
+
+const C = 2
+
+type I interface {
+	F() [C]int
```
Comment (codyoss) [inline mockgen/internal/tests/const_array_length/input.go:7]: Would you add another function to test the other case for arrays: `Bar() [2]int`

File: mockgen/parse_test.go [hunk 3]
```diff
@@ -263,3 +263,27 @@ func TestParsePackageImport_FallbackMultiGoPath(t *testing.T) {
...
+
+	pkg, err := p.parseFile("", file)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	expect := "[2]int"
+	got := pkg.Interfaces[0].Methods[0].Out[0].Type.String(nil, "")
+	if got != expect {
+		t.Fatalf("Expected return type %v but got %v", expect, got)
```
Comment (codyoss) [inline mockgen/parse_test.go:287]: nit: Could you reverse this message so it is in the form of "got %v, want %v". Again I know this is not consistent with some of the other tests, but they should be refactored. See [useful test failures](https://gist.github.com/adamveld12/c0d9f0d5f0e1fba1e551#useful-test-failures).

**Review by codyoss — CHANGES_REQUESTED** [review]
Just a couple of small things, thanks for the contribution!

**Comment by fsmiamoto** [comment]
Thanks for the review! I guess we should be good now.

**Review by codyoss — APPROVED** [review]

**Comment by codyoss** [comment]
Thank you for your contribution! 🎆

**Pull request merged**